backend/de.metas.monitoring/src/main/java/de/metas/monitoring/adapter/MicrometerPerformanceMonitoringService.java

```diff
@@ -42,63 +42,88 @@
 public class MicrometerPerformanceMonitoringService implements PerformanceMonitoringService
 {
 	private final MeterRegistry meterRegistry;
-	private final Optional<APMPerformanceMonitoringService> apmPerformanceMonitoringService;
+	private static final ThreadLocal<PerformanceMonitoringData> perfMonDataTL = ThreadLocal.withInitial(PerformanceMonitoringData::new);
 
 	public MicrometerPerformanceMonitoringService(
-			@NonNull final Optional<APMPerformanceMonitoringService> apmPerformanceMonitoringService,
 			@NonNull final MeterRegistry meterRegistry)
 	{
-		this.apmPerformanceMonitoringService = apmPerformanceMonitoringService;
 		this.meterRegistry = meterRegistry;
 	}
 
 	@Override
-	public <V> V monitorTransaction(
+	public <V> V monitor(
 			@NonNull final Callable<V> callable,
-			@NonNull final TransactionMetadata metadata)
+			@NonNull final PerformanceMonitoringService.Metadata metadata)
 	{
-		final ArrayList<Tag> tags = createTagsFromLabels(metadata.getLabels());
-		mkTagIfNotNull("Name", metadata.getName()).ifPresent(tags::add);
 
-		final Callable<V> callableToUse;
-		if (apmPerformanceMonitoringService.isPresent())
+		final PerformanceMonitoringData perfMonData = perfMonDataTL.get();
+		final ArrayList<Tag> tags = new ArrayList<>();
+
+		mkTagIfNotNull("name", metadata.getClassName()).ifPresent(tags::add);
+		mkTagIfNotNull("action", metadata.getFunctionName()).ifPresent(tags::add);
+
```

Comment:
don't understand the cases from the if/else  below.

some random thoughts:
* aren't REST_CONTROLLER_WITH_WINDOW_ID and REST_CONTROLLER quite similar
* instead of having whole mambo-jambo logics like 
```
metadata.getType() == Type.REST_CONTROLLER_WITH_WINDOW_ID
				&& metadata.getFunctionName() != null
				&& metadata.getWindowNameAndId() != null)
				````
couldn't we have a nice method into metadata which have a descriptive name?